### PR TITLE
remove spurious error meesage

### PR DESCRIPTION
**Pull request by ChrisDodd**

### CODE REVIEW

File: frontends/p4/fromv1.0/programStructure.cpp [hunk 1]
```diff
@@ -285,12 +285,6 @@ class ProgramStructure::FixupExtern : public Modifier {
 };
 
 void ProgramStructure::createExterns() {
-    for (auto r : registers) {
```
Comment (mbudiu-vmw) [inline frontends/p4/fromv1.0/programStructure.cpp]: Well, the back-end does not need to generate code for direct registers anywhere.
Where are they supported?

Comment (ChrisDodd) [inline frontends/p4/fromv1.0/programStructure.cpp]: They're supported by other backends, just not by the bmv2 backend.

Comment (mbudiu-vmw) [inline frontends/p4/fromv1.0/programStructure.cpp]: But also they do not seem to be converted to P4-16 data structures anywhere.
Where is that done?

Comment (mbudiu-vmw) [inline frontends/p4/fromv1.0/programStructure.cpp]: I don't see where they generate P4-16 code.

Comment (mbudiu-vmw) [inline frontends/p4/fromv1.0/programStructure.cpp]: Ok, they generate declarations, but the declarations will be removed as dead code, as they are not referenced anywhere. 

Also, if you remove this error message at least one test will fail.

Comment (mbudiu-vmw) [inline frontends/p4/fromv1.0/programStructure.cpp]: Maybe you can consider moving this error message to the BMv2 back-end?

Comment (ChrisDodd) [inline frontends/p4/fromv1.0/programStructure.cpp]: Moved the check to the bmv2 backend

**Pull request merged**